docs/src/Fieldset.doc.js

```diff
@@ -0,0 +1,222 @@
+                  />
+                  <RadioButton
+                    checked={favorite === 'plants'}
+                    id="favoritePlantsA11y"
+                    label="Plants"
+                    name="favorite"
+                    onChange={() => setFavorite( 'plants' )}
+                    value="plants"
+                  />
+                </Flex>
+              </Fieldset>
+            </Flex>
+          </form>
+        );
+      }
+`}
+      />
+    </MainSection.Subsection>
+  </MainSection>,
+);
+
+card(<MainSection name="Localization" description={`Be sure to localize the \`legend\` text.`} />);
+
+card(
+  <MainSection name="Variants">
+    <MainSection.Subsection
+      description={`Be default, the \`legend\` is visible above the items in the Fieldset. However, if the form items are labelled by content elsewhere on the page, or a more complex legend is needed, the \`legendDisplay\` prop can be used to visually hide the legend. In this case, it is still available to screen reader users, but will not appear visually on the screen.`}
+      title="Legend visibility"
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+function CheckboxExample() {
+  const [checkedBotanical, setCheckedBotanical] = React.useState(false);
+  const [checkedTypo, setCheckedTypo] = React.useState(false);
+
+  return (
+    <Flex direction="column" gap={4}>
+      <Box>
+        <Text weight="bold">Design options</Text>
+        <Text size="md">Choose which backgrounds to use as placeholders when creating content</Text>
+      </Box>
+      <Fieldset legend="Which backgrounds would you like to use?" legendDisplay="hidden">
+        <Flex direction="column" gap={4}>
+          <Checkbox
+            checked={checkedBotanical}
+            id="coral"
+            label="Coral"
+            subtext="Botanical art in coral and green"
+            image={<Box height={100} width={80}><Image alt="Botanical art in coral and green" src="https://i.ibb.co/7bQQYkX/stock2.jpg" fit="contain" naturalWidth={1} naturalHeight={1}/></Box>}
+            name="favorite art"
+            onChange={({ checked }) => {
+              setCheckedBotanical(checked);
+            }}
+          />
+          <Checkbox
+            checked={checkedTypo}
+            id="blue"
+            label="Blue"
+            subtext="Typography and shoe in blue"
+            image={<Box height={100} width={80}><Image alt="Typography and shoe in blue" src="https://i.ibb.co/jVR29XV/stock5.jpg" fit="contain" naturalWidth={1} naturalHeight={1}/></Box>}
+            name="favorite art"
+            onChange={({ checked }) => {
+              setCheckedTypo(checked);
+            }}
+          />
+        </Flex>
+      </Fieldset>
+    </Flex>
+  );
+}
+`}
+      />
+    </MainSection.Subsection>
+  </MainSection>,
+);
+
+card(
+  <MainSection name="Related">
+    <MainSection.Subsection
+      description={`
+      **[Label](/Label)**
```

Comment:
Should we add a best practice do/don't explicitly? "Don't use Fieldset for single form elements. Use Label or the build-in label prop on them."

Reply:
We can add this in the next iteration when we add some styling options too